main.cpp

```diff
@@ -224,28 +124,63 @@ struct HeapA
 #include <cmath>
 #include <functional>
 #include <memory>
+#include <typeinfo>
+#include <utility>
+
+template<typename NumericType>
+struct Temporary
+{
+    Temporary(NumericType t) : v(t)
+    {
+        std::cout << "I'm a Temporary<" << typeid(v).name() << "> object, #"
+                  << counter++ << std::endl;
+    }
+    /*
+     revise these conversion functions to read/write to 'v' here
+     hint: what qualifier do read-only functions usually have?
+     */
+    operator NumericType() const { return v;}
+    operator NumericType&() { return v;}
+    
+private:
+    static int counter;
+    NumericType v;
+};
+
+template<typename NumericType>
+int Temporary<NumericType>::counter = 0;
 
 template <typename NumericType>
 struct Numeric
 {
     using Type = NumericType;
-    explicit Numeric(Type v) : value(std::make_unique<Type>(v)) {}
+    explicit Numeric(Type v) : value(std::make_unique<Temporary<Type>>(v)) {}
```

Comment:
```
 Your job is to replace the owned type (the primitive specified by your template argument) 
from the Templated Class you created in Ch5 p04 with a struct named 
Temporary that can behave ENTIRELY as a temporary object.
```
you have the wrong typename on the right side of the `=` sign.

your `unique_ptr<>` owns `Type`, not `Temporary<Type>` 

Change the `using` alias, not the constructor.

- [ ] revise